var/spack/repos/builtin/packages/py-gmxapi/package.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import os
+
+from spack.package import *
+
+
+class PyGmxapi(PythonPackage):
+    """Python bindings and ensemble workflow management for GROMACS.
+
+    The GROMACS C++ API is affected by its package variants. You can
+    specify a particular GROMACS API by making the dependency variant explicit.
+    E.g. ``spack install gmxapi ^gromacs+mpi~double``
+    """
+
+    homepage = "https://manual.gromacs.org/current/gmxapi/index.html"
+    maintainers("eirrgang", "peterkasson")
+
+    pypi = "gmxapi/gmxapi-0.4.0.tar.gz"
+    version("0.4.0", sha256="7fd58e6a4b1391043379e8ba55555ebeba255c5b394f5df9d676e6a5571d7eba")
+
+    depends_on("gromacs~mdrun_only+shared")
+    depends_on("mpi")
+    depends_on("py-cmake@3.16:", type="build")
+    depends_on("py-importlib-metadata", type="test", when="^python@:3.7")
+    depends_on("py-mpi4py", type=("build", "run"))
+    depends_on("py-networkx@2.0:", type=("build", "run"))
+    depends_on("py-numpy@1.7:", type=("build", "run"))
+    depends_on("py-setuptools@42:", type="build")
+    depends_on("py-packaging", type=("build", "run"))
+    depends_on("py-pybind11@2.6:", type="build")
+    depends_on("py-pytest@4.6:", type="test")
+    depends_on("py-wheel", type="build")
+
+    conflicts("^gromacs@:2022.0", when="@0.4:", msg="Use GROMACS 2022.1 or newer for gmxapi 0.4.")
+    conflicts(
+        "^gromacs+mpi", when="@:0.3.2", msg="Use gmxapi 0.4 or higher for MPI-enabled GROMACS."
+    )
```

Comment:
Per @adamjstewart : Don't bother with (second-class) support for older versions. 
```suggestion
    depends_on("gromacs@2022.1:~mdrun_only+shared")
    depends_on("mpi")
    depends_on("py-cmake@3.16:", type="build")
    depends_on("py-importlib-metadata", type="test", when="^python@:3.7")
    depends_on("py-mpi4py", type=("build", "run"))
    depends_on("py-networkx@2.0:", type=("build", "run"))
    depends_on("py-numpy@1.7:", type=("build", "run"))
    depends_on("py-setuptools@42:", type="build")
    depends_on("py-packaging", type=("build", "run"))
    depends_on("py-pybind11@2.6:", type="build")
    depends_on("py-pytest@4.6:", type="test")
    depends_on("py-wheel", type="build")
```